mozci/push.py

```diff
@@ -533,6 +535,39 @@ def scheduled_duration(self):
 
         return int(duration / 3600)
 
+    def is_group_running(self, group):
+        """Checks if the provided group is still running on this push.
+
+        Returns:
+            bool: True, if the group is still running.
+                  False, if the group is completed.
+        """
+        running_tasks = [
+            task
+            for task in self.tasks
+            if task.state not in ["completed", "failed", "exception"]
+        ]
+        if all(task.is_tests_grouped() for task in group.tasks):
+            for task in running_tasks:
+                task_def = get_task(task.id)
+                test_paths = json.loads(
+                    task_def["payload"].get("MOZHARNESS_TEST_PATHS", "{}")
+                )
+                if group.name in set(
+                    [name for names in test_paths.values() for name in names]
+                ):
+                    return True
+        else:
+            group_types = set([task.tags.get("tests-type", "") for task in group.tasks])
+            running_types = set(
+                [task.tags.get("tests-type", "") for task in running_tasks]
+            )
+            return any(
+                tests_type in running_types for tests_type in group_types if tests_type
+            )
```

Comment:
Nit: this could be simplified by doing something like:
```Python
return not group_types.isdisjoint(running_types)
```
or
```Python
return len(group_types | running_types) != 0
```